packages/Meshfree/test/tstMeshfreeComparison.cpp

```diff
@@ -0,0 +1,271 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <DTK_MovingLeastSquaresOperator.hpp>
+#include <DTK_SplineOperator.hpp>
+
+constexpr int DIM = 3;
+
+template <typename DeviceType>
+struct Helper
+{
+    static std::vector<std::array<double, DIM>>
+    makeGridPoints( std::array<int, DIM> const &n_points,
+                    std::array<double, DIM> const &lower_corner,
+                    std::array<double, DIM> const &upper_corner )
+    {
+        static_assert( DIM == 3, "Assume three dimensional geometry" );
+
+        std::vector<std::array<double, DIM>> grid_points;
+
+        for ( int i = 0; i < n_points[0]; ++i )
+        {
+            for ( int j = 0; j < n_points[1]; ++j )
+            {
+                for ( int k = 0; k < n_points[2]; ++k )
+                {
+                    std::array<double, DIM> point = {
+                        lower_corner[0] +
+                            ( upper_corner[0] - lower_corner[0] ) /
+                                ( n_points[0] + 1 ) * ( i + 1 ),
+                        lower_corner[1] +
+                            ( upper_corner[1] - lower_corner[1] ) /
+                                ( n_points[1] + 1 ) * ( j + 1 ),
+                        lower_corner[2] +
+                            ( upper_corner[2] - lower_corner[2] ) /
+                                ( n_points[2] + 1 ) * ( k + 1 )};
+                    grid_points.push_back( point );
```

Comment:
Hmm... AFAICT I get `[lower_corner+eps, upper_corner-eps]` with `eps=upper_corner-lower_corner/(n_points+1)`. I wanted to avoid using duplicated points.